src/Console.cc

```diff
@@ -14,223 +14,74 @@
-// They happen to overlap, but there might be differences if more colors are
-// added.
-const int red = 31;
-const int yellow = 33;
-const int green = 32;
-const int blue = 36;
-
-Logger Console::err("[Err] ", red, Logger::STDERR, 1);
-Logger Console::warn("[Wrn] ", yellow, Logger::STDERR, 2);
-Logger Console::msg("[Msg] ", green, Logger::STDOUT, 3);
-Logger Console::dbg("[Dbg] ", blue, Logger::STDOUT, 4);
-
-int Console::verbosity = 1;
-std::string Console::customPrefix = ""; // NOLINT(*)
-
-//////////////////////////////////////////////////
-void Console::SetVerbosity(const int _level)
-{
-  verbosity = _level;
-}
-
-//////////////////////////////////////////////////
-int Console::Verbosity()
-{
-  return verbosity;
-}
-
-//////////////////////////////////////////////////
-void Console::SetPrefix(const std::string &_prefix)
-{
-  customPrefix = _prefix;
-}
-
-//////////////////////////////////////////////////
-std::string Console::Prefix()
-{
-  return customPrefix;
-}
-
-/////////////////////////////////////////////////
-Logger::Logger(const std::string &_prefix, const int _color,
-               const LogType _type, const int _verbosity)
-: std::ostream(new Buffer(_type, _color, _verbosity)), prefix(_prefix)
-{
-  this->setf(std::ios_base::unitbuf);
-}
-
-/////////////////////////////////////////////////
-Logger::~Logger()
-{
-}
-
-/////////////////////////////////////////////////
-Logger &Logger::operator()()
-{
-  Console::log() << "(" << common::systemTimeIso() << ") ";
-  (*this) << Console::Prefix() << this->prefix;
-
-  return (*this);
-}
-
-/////////////////////////////////////////////////
-Logger &Logger::operator()(const std::string &_file, int _line)
-{
-  int index = _file.find_last_of("/") + 1;
+#include <filesystem>
+#include <iostream>
+#include <sstream>
+#include <string>
 
-  Console::log() << "(" << common::systemTimeIso() << ") ";
-  std::stringstream prefixString;
-  prefixString << Console::Prefix() << this->prefix
-    << "[" << _file.substr(index , _file.size() - index) << ":"
-    << _line << "] ";
-  (*this) << prefixString.str();
+#include <gz/common/config.hh>
+#include <gz/common/Console.hh>
+#include <gz/common/Util.hh>
+#include <gz/utils/log/Logger.hh>
+#include <gz/utils/NeverDestroyed.hh>
 
-  return (*this);
-}
+using namespace gz;
+using namespace common;
 
 /////////////////////////////////////////////////
-Logger::Buffer::Buffer(LogType _type, const int _color, const int _verbosity)
-  :  type(_type), color(_color), verbosity(_verbosity)
+LogMessage::LogMessage(const char *_file, int _line,
+  spdlog::level::level_enum _logLevel, bool _fileInitialize)
+  : severity(_logLevel),
+    sourceLocation(_file, _line, "")
 {
+  // Use default initialization if needed.
+  if (_fileInitialize && !Console::initialized)
+  {
+    Console::Init(".gz", "auto_default.log");
+    // Allow gzlog to pass through.
+    Console::SetVerbosity(5);
```

Comment:
Do we want to set the verbosity to 5 here? On line 66, I see `Console::verbosity = 1`.

Reply:
Fixed in https://github.com/gazebosim/gz-common/pull/615/commits/f3d168cfdc7eb4a8575edc4dfbf24cd8a98a5c52

Reply:
Removed this changed in [f3d168c](https://github.com/gazebosim/gz-common/pull/615/commits/f3d168cfdc7eb4a8575edc4dfbf24cd8a98a5c52).